### PR TITLE
Add edit survey

### PR DESCRIPTION
**Pull request by sohilpandya**
This contains all the changes made by Franz. Some of these are refactored/edited by me in a future PR. 

Review this first

### CODE REVIEW

File: server/lib/composeQuestionStatement.js [hunk 1]
```diff
@@ -4,14 +4,21 @@
...
     for (var i = 0; i < rows.length; i++) {
         var row = rows[i];
+        if (isSurvey) {
+            row.survey_id = id;
+        } else {
+            row.quiz_id = id;
+        }
         var valueClause = [];
-        params.push(row.quiz_id);
+        params.push(row[quizIdOrSurveyId]);
```
Comment (Danwhy) [inline server/lib/composeQuestionStatement.js]: Couldn't we just push in the id here, and not bother with lines 15 - 19?

File: server/lib/getModuleForLecturer.js [hunk 2]
```diff
@@ -16,52 +16,65 @@ var organiseModuleData = require('./organiseModuleData');
...
+                                            return callback(error);
+                                        }
+                                        const allData = {
+                                            quizzes: quizzes.rows,
+                                            surveys: surveys.rows,
+                                            medals: medals.rows,
+                                            trophies: trophies.rows,
+                                            num_enrolled: num_enrolled.rows,
+                                            name: name.rows
+                                        };
```
Comment (Danwhy) [inline server/lib/getModuleForLecturer.js]: https://github.com/cul-2016/quiz/issues/325#issuecomment-273749798

Not a priority, but something to look into/ think about

File: server/plugins/quizes.js [hunk 3]
```diff
@@ -218,8 +228,9 @@ exports.register = (server, options, next) => {
                     const { quiz_id, score } = request.query;
                     const { user_id } = decoded.user_details;
                     if (quiz_id !== undefined && score !== undefined) {
-
-                        editScore(pool, user_id, quiz_id, score, (error, response) => {
+                        const parsed_quiz_id = parseInt(quiz_id, 10);
+                        const parsed_score = parseInt(quiz_id, 10);
```
Comment (Danwhy) [inline server/plugins/quizes.js]: should this be `const parsed_score = parseInt(score, 10);`?

**Comment by sohilpandya** [comment]
Thanks for the review @Danwhy , 2 of the above points needed to be addressed :)

Passing back to you. :)

**Pull request merged**